### PR TITLE
Add messages when converting to protobuf version of block data

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

This PR is needed to properly submit messages, and fixes the "bug" where the chain halts after submitting a signed pay for message.

### CODE REVIEW

File: types/block.go [hunk 1]
```diff
@@ -1251,14 +1247,24 @@ func (data *Data) ToProto() tmproto.Data {
...
+	protoMsgs := make([]*tmproto.Message, len(data.Messages.MessagesList))
+	for i, msg := range data.Messages.MessagesList {
+		protoMsgs[i] = &tmproto.Message{
+			NamespaceId: msg.NamespaceID,
+			Data:        msg.Data,
+		}
+	}
+	tp.Messages = tmproto.Messages{MessagesList: protoMsgs}
+
 	return *tp
```
Comment (Wondertan) [inline types/block.go:1268]: Yeaah, you found it. Congrats. LGTM

**Review by Wondertan — APPROVED** [review]

File: types/block.go [hunk 2]
```diff
@@ -1251,14 +1247,24 @@ func (data *Data) ToProto() tmproto.Data {
...
-	// TODO(evan): copy missing pieces from previous work
+
 	pevd, err := data.Evidence.ToProto()
 	if err != nil {
 		// TODO(evan): fix
 		panic(err)
 	}
 	tp.Evidence = *pevd
 
+	// need to add messages
```
Comment (adlerjohn) [inline types/block.go:1258]: This comment makes it sound like a TODO?

**Review by adlerjohn — COMMENTED** [review]
Is it expected that the test fails?

**Review by liamsi — APPROVED** [review]
How did we not catch this before? 😅 
Great find, straightforward fix!

File: types/block.go (hunk 2 above)
Comment (evan-forbes) [reply types/block.go]: good point, we should remove this https://github.com/celestiaorg/celestia-core/pull/580/commits/7f053bb043de85efeb8f58a6a431d47d5eff7247

**Pull request merged**